source/common/http/conn_manager_impl.cc

```diff
@@ -2742,12 +2790,7 @@ ConnectionManagerImpl::ActiveStreamDecoderFilter::getUpstreamSocketOptions() con
 
 void ConnectionManagerImpl::ActiveStreamDecoderFilter::requestRouteConfigUpdate(
     Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb) {
-  parent_.active_stream_.requestRouteConfigUpdate(dispatcher(), std::move(route_config_updated_cb));
-}
-
-absl::optional<Router::ConfigConstSharedPtr>
-ConnectionManagerImpl::ActiveStreamDecoderFilter::routeConfig() {
-  return parent_.active_stream_.routeConfig();
+  parent_.active_stream_.requestRouteConfigUpdate(route_config_updated_cb);
```

Comment:
Do you want to be issuing an SRDS config request every time this is invoked?

Reply:
I guess so, we don't know whether the route table has been initialized when we are in the worker thread. A callback must be posted to the main thread to request the update. This is invoked only when route is not found and on_demand_filter is configured.